ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
+        }
+
         $button = array(
           array(
             '#theme' => 'link',
             '#text' => t('Reserve'),
-            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . implode('/', $path),
+            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . base64_encode(implode('|', $item_id)),
```

Comment:
Encoding the values separately does not increase length or weirdness.

[A quick test](https://3v4l.org/bric0):

```php
$item_id = array('49408625ZX06-07ZXZX2015ZXZX', '2015', '06-07');
implode('/', array_map('base64_encode', $item_id)) == "NDk0MDg2MjVaWDA2LTA3WlhaWDIwMTVaWFpY/MjAxNQ==/MDYtMDc=";
base64_encode(implode('|', $item_id)) == "NDk0MDg2MjVaWDA2LTA3WlhaWDIwMTVaWFpYfDIwMTV8MDYtMDc=";
```